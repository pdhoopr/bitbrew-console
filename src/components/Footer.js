import { observer } from 'mobx-react';
import React from 'react';
import styled from 'styled-components';
import Link from '../styles/Link';

const Wrapper = styled.footer`
  margin-bottom: var(--size-32);
  margin-top: var(--size-128);
`;

const List = styled.ul`
  color: var(--color-dark-grey);
  list-style-type: none;
  margin-bottom: 0;
  margin-top: 0;
  padding-left: 0;
  text-align: center;
`;

const ListItem = styled.li`
  display: inline-block;

  &:not(:last-of-type) {
    margin-right: var(--size-24);
  }
`;

function Footer() {
  return (
    <Wrapper>
      <List>
        <ListItem>&copy; 2018 BitBrew, Inc.</ListItem>
        <ListItem>
          <Link
            href="http://docs.hub.bitbrew.com/v1.3/reference#authentication"
            rel="noopener noreferrer"
            target="_blank"
          >
            API Documentation
          </Link>
        </ListItem>
        <ListItem>
          <Link
            href="http://docs.hub.bitbrew.com/v1.3/reference"
            rel="noopener
            noreferrer"
            target="_blank"
          >
            Event Catalog
          </Link>
        </ListItem>
        <ListItem>
          <Link
            href="https://help.bitbrew.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Help
          </Link>
        </ListItem>
      </List>
    </Wrapper>
  );
}

export default observer(Footer);
