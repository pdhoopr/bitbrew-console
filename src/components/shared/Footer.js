import React from "react";
import styled from "styled-components";
import { Link } from "../../design-system";

const Docs = styled.ul`
  align-items: center;
  color: var(--color-dark-gray);
  display: flex;
  justify-content: center;
  list-style-type: none;
  margin-bottom: 0;
  margin-top: 0;
  padding: var(--size-96) var(--size-24) var(--size-32);
`;

const DocsLink = styled(Link)`
  margin-left: var(--size-24);
`;

export default function Footer() {
  return (
    <footer>
      <Docs>
        <li>&copy; 2019 BitBrew, Inc.</li>
        <li>
          <DocsLink to="https://help.bitbrew.com/" isExternal>
            Help
          </DocsLink>
        </li>
        <li>
          <DocsLink to="https://docs.hub.bitbrew.com/reference" isExternal>
            Reference
          </DocsLink>
        </li>
      </Docs>
    </footer>
  );
}
