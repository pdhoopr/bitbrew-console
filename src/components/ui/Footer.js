import { observer } from 'mobx-react';
import React from 'react';
import styled from 'styled-components';
import { Link } from './Links';

const Wrapper = styled.footer`
  margin-bottom: var(--size-32);
  margin-top: var(--size-128);
`;

const Resources = styled.ul`
  color: var(--color-dark-gray);
  list-style-type: none;
  margin-bottom: 0;
  margin-top: 0;
  padding-left: 0;
  text-align: center;
`;

const Item = styled.li`
  display: inline-block;

  &:not(:last-of-type) {
    margin-right: var(--size-16);
  }
`;

function Footer() {
  return (
    <Wrapper>
      <Resources>
        <Item>&copy; 2018 BitBrew, Inc.</Item>
        <Item>
          <Link to="http://docs.hub.bitbrew.com/v1.3/reference#authentication">
            API Documentation
          </Link>
        </Item>
        <Item>
          <Link to="http://docs.hub.bitbrew.com/v1.3/reference">
            Event Catalog
          </Link>
        </Item>
        <Item>
          <Link to="https://help.bitbrew.com/">Help</Link>
        </Item>
      </Resources>
    </Wrapper>
  );
}

export default observer(Footer);
