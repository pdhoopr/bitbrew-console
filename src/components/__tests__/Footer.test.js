import { shallow } from 'enzyme';
import React from 'react';
import Footer from '../Footer';

test('renders DOM for the footer', () => {
  const dom = shallow(<Footer />);
  expect(dom).toMatchSnapshot();
});
