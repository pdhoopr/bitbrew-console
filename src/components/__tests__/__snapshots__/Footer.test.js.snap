// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DOM for the footer 1`] = `
<styled.footer>
  <styled.ul>
    <styled.li>
      Â© 2018 BitBrew, Inc.
    </styled.li>
    <styled.li>
      <styled.a
        href="http://docs.hub.bitbrew.com/v1.3/reference#authentication"
        rel="noopener noreferrer"
        target="_blank"
      >
        API Documentation
      </styled.a>
    </styled.li>
    <styled.li>
      <styled.a
        href="http://docs.hub.bitbrew.com/v1.3/reference"
        rel="noopener noreferrer"
        target="_blank"
      >
        Event Catalog
      </styled.a>
    </styled.li>
    <styled.li>
      <styled.a
        href="https://help.bitbrew.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Help
      </styled.a>
    </styled.li>
  </styled.ul>
</styled.footer>
`;
